src/java/org/apache/cassandra/db/lifecycle/View.java

```diff
@@ -353,4 +361,40 @@ public boolean apply(T t)
             }
         };
     }
+
+    // Match the SSTableReaders from the existing ones to the new one to be added (with same ranges)
+    // Returns the map of toRemove <-> toAdd. Return empty map if such 1-1 replacement doesn't exist
+    private static Map<SSTableReader, SSTableReader> getReplacementMap(final Set<SSTableReader> remove, final Iterable<SSTableReader> add)
+    {
+        List<SSTableReader> toAdds = new ArrayList<>();
```

Comment:
The copy and sorting of `toAdds` and `toRemove` could be done outside the lock/CAS loop since it doesn't depend on the contents of the `View`